portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,24 +35,37 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Youtube Idea").addSort("timestamp", SortDirection.ASCENDING);
 
-    // TODO:  Make ArrayList more abstract aka make it a List.
-    ArrayList<String> sampleContents = new ArrayList<String>();
-    sampleContents.add("My name is Miah");
-    sampleContents.add("Im from NYC");
-    sampleContents.add("I love coding!");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> ideas = new ArrayList<>();
+    for (Entity entity: results.asIterable()) {
+        String idea = (String) entity.getProperty("idea");
+
+        ideas.add(idea);
+    }
+
+    Gson gson = new Gson();
 
-    // Send the JSON as the response
-    String json = convertToJsonUsingGson(sampleContents);
-    String jsonComment = new Gson().toJson(sampleContents);
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(gson.toJson(ideas));
+
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      String userInput = request.getParameter("user-input");
-      response.setContentType("text/html;");
-      response.getWriter().println("Thanks for the idea! If I ever make a video about " + userInput + " you'll be the first to know");
+      String userIdea = request.getParameter("user-input");
+      long timestamp = System.currentTimeMillis();
+
+      Entity youtubeEntity = new Entity("Youtube Idea");
+      youtubeEntity.setProperty("idea", userIdea);
+      youtubeEntity.setProperty("timestamp", timestamp);
+      
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();      
```

Comment:
this line can be removed once you've made this a private final instance variable